lib/MySQL_Session.cpp

```diff
@@ -6414,6 +6415,54 @@ bool MySQL_Session::handler___status_WAITING_CLIENT_DATA___STATE_SLEEP___MYSQL_C
 							proxy_debug(PROXY_DEBUG_MYSQL_COM, 8, "Changing connection charset to %d\n", c->nr);
 							client_myds->myconn->set_charset(c->nr, NAMES);
 						}
+					} else if (var == "wait_timeout") {
+						// MySQL min allowed value, 1 second in milliseconds
+						const unsigned long long MYSQL_WAIT_TIMEOUT_MIN_MS = 1 * 1000;
+						// MySQL max allowed value, 365 days in milliseconds
+						const unsigned long long MYSQL_WAIT_TIMEOUT_MAX_MS = 31536000ULL * 1000;
+
+						std::string value = *values++;
+						proxy_debug(PROXY_DEBUG_MYSQL_COM, 5, "Client requested SET wait_timeout = %s\n", value.c_str());
+
+						unsigned long long client_timeout = 0;
+						try {
+							client_timeout = std::stoull(value) * 1000;
+						} catch (const std::exception& e) {
```

Comment:
I don't think input validation should be done here.
Numeric variables (like `wait_timeout`) should be defined in `mysql_variables_numeric` , and marked as numeric.
IIRC , the parser should take care of validating that the value is indeed numeric.